preflight.inc.php

```diff
@@ -142,14 +142,21 @@
 		$errors++;
 	}
 
-	if(in_array('mod_rewrite', apache_get_modules())){
-		$tests['mod_rewrite']['state']="good";
-		$tests['mod_rewrite']['message']='mod_rewrite detected';
-		$tests['api_test']['state']="fail";
-		$tests['api_test']['message']="Apache does not appear to be rewriting URLs correctly. Check your AllowOverride directive and change to 'AllowOverride All'";
-	}else{
-		$tests['mod_rewrite']['state']="fail";
-		$tests['mod_rewrite']['message']='Apache is missing the <a href="http://httpd.apache.org/docs/current/mod/mod_rewrite.html">mod_rewrite</a> module and it is required for the API to function correctly.  Please install it.';
+    if (function_exists('apache_get_modules')) {
+        if (in_array('mod_rewrite', apache_get_modules())) {
+            $tests['mod_rewrite']['state']="good";
+            $tests['mod_rewrite']['message']='mod_rewrite detected';
+            $tests['api_test']['state']="fail";
+            $tests['api_test']['message']="Apache does not appear to be rewriting URLs correctly. Check your AllowOverride directive and change to 'AllowOverride All'";
+        } else{
+            $tests['mod_rewrite']['state']="fail";
+            $tests['mod_rewrite']['message']='Apache is missing the <a href="http://httpd.apache.org/docs/current/mod/mod_rewrite.html">mod_rewrite</a> module and it is required for the API to function correctly.  Please install it.';
+        }
+    } else {
+        $tests['mod_rewrite']['state']="good";
+        $tests['mod_rewrite']['message']='assume mod_rewrite is enabled';
+		$tests['api_test']['state']="good";
```

Comment:
the api test is supposed to be in a fail state until the javascript loads on the client screen and then alters it to say good that the test passed.